symengine/eval_mpfr.cpp

```diff
@@ -241,6 +241,24 @@ class EvalMPFRVisitor : public BaseVisitor<EvalMPFRVisitor> {
         x.eval(mpfr_get_prec(result_))->accept(*this);
     }
 
+    void bvisit(const Max &x) {
+        apply(result_, *(x.get_args()[0]));
+        mpfr_ptr tmp = result_;
```

Comment:
This is just pointer assignment and `tmp` now points to what is pointed by `result_`. You should initialize a new variable mpfr_t.